src/middleware.js

```diff
@@ -38,6 +40,45 @@ const expressMiddleware = ({
   }
 }
 
+/**
+ * Generates a request tracer plugin for Fastify.
+ *
+ * @param {Object} options possible options
+ * @param {boolean} options.useHeader respect request header flag
+ *                                    (default: `false`)
+ * @param {string} options.headerName request header name, used if `useHeader` is set to `true`
+ *                                    (default: `X-Request-Id`)
+ * @param {boolean} options.useFastifyRequestId respect Fastify request id flag
+ *                                    (default: `false`)
+ */
+const fastifyPlugin = ({
```

Comment:
options should be passed to `register`, not the plugin directly. So you'd get the options as the arg you have as `_` now

Reply:
Do you think it makes sense to change it? The plugin doesn't expect additional options provided by fastify, so the only difference I see is consistency with other fastify plugins. Also, it makes sense to get rid of the factory function along with the proposed change and have a single plugin instance.

Reply:
>so the only difference I see is consistency with other fastify plugins

I think that's a valuable goal :)

Reply:
This change will happen in #34